### PR TITLE
JohnPapa and documentation

### PR DESCRIPTION
**Pull request by Atap23**
Features:

 - JohnPaparization.
 - Added some documentation.

### CODE REVIEW

File: source/_shared/source._shared.model.provider.js [hunk 1]
```diff
@@ -72,7 +60,7 @@
 
     return {
       get: getProvider,
-      $get: [$get]
+      $get: $get
```
Comment (nulpas) [inline source/_shared/source._shared.model.provider.js]: Return brackets

File: source/_shared/source._shared.tools.provider.js [hunk 2]
```diff
@@ -22,7 +22,7 @@
       $: $,
       /* Object tools */
       setObjectUsingSchema: setObjectUsingSchemaProvider,
-      $get: [$get]
+      $get: $get
```
Comment (nulpas) [inline source/_shared/source._shared.tools.provider.js]: Return brackets

File: source/api/source.api.model.provider.js [hunk 3]
```diff
@@ -153,6 +153,18 @@
                   $top: 30
                 }
               },
+              /**
+               * @name QUERY.subProcess
```
Comment (nulpas) [inline source/api/source.api.model.provider.js]: Heritage

File: source/api/source.api.model.provider.js [hunk 4]
```diff
@@ -153,6 +153,18 @@
...
+               * @memberof source.api.apiModelProvider
+               *
+               * @description
+               * It makes a "GET" Api call through Restangular. If the requestObject provided has an entity that have
+               * an entityId, it will make a call to retrieve that concrete entity. Otherwise, it will make a call
+               * to retrieve all existing entities.
+               *
+               * @param requestObject
```
Comment (nulpas) [inline source/api/source.api.model.provider.js]: Param type

File: source/api/source.api.model.provider.js [hunk 5]
```diff
@@ -167,6 +179,16 @@
                 defaultHeaders: _defaults.defaultHeaders,
                 defaultParams: _defaults.defaultParams
               },
+              /**
+               * @name LOCAL.subProcess
```
Comment (nulpas) [inline source/api/source.api.model.provider.js]: Heritage

File: source/api/source.api.model.provider.js [hunk 6]
```diff
@@ -167,6 +179,16 @@
...
+               * @memberof source.api.apiModelProvider
+               *
+               * @description
+               * It makes a local "GET" API call through Restangular.
+               *
+               * @param requestObject
```
Comment (nulpas) [inline source/api/source.api.model.provider.js]: Param type

File: source/router/source.router.factory.js [hunk 7]
```diff
@@ -1,29 +1,55 @@
...
+     */
+    function _resolveStateGo(stateName) {
+      $timeout(function() {
+        $state.go(stateName);
+      });
+    }
+
+    /**
+     * @name resolveStateGo
+     * @memberof source.router.$routerProvider
```
Comment (nulpas) [inline source/router/source.router.factory.js]: $router

File: source/router/source.router.service.js [hunk 8]
```diff
@@ -1,17 +1,26 @@
 (function() {
```
Comment (nulpas) [inline source/router/source.router.service.js]: Delete this file.

File: source/_shared/source._shared.model.provider.js (hunk 1 above)
Comment (Atap23) [inline source/_shared/source._shared.model.provider.js]: Done.

File: source/_shared/source._shared.tools.provider.js (hunk 2 above)
Comment (Atap23) [inline source/_shared/source._shared.tools.provider.js]: Donde

File: source/api/source.api.model.provider.js (hunk 3 above)
Comment (Atap23) [inline source/api/source.api.model.provider.js]: Done

File: source/api/source.api.model.provider.js (hunk 4 above)
Comment (Atap23) [inline source/api/source.api.model.provider.js]: Done

File: source/api/source.api.model.provider.js (hunk 5 above)
Comment (Atap23) [inline source/api/source.api.model.provider.js]: Done

File: source/api/source.api.model.provider.js (hunk 6 above)
Comment (Atap23) [inline source/api/source.api.model.provider.js]: Done.

File: source/router/source.router.factory.js [hunk 9]
```diff
@@ -1,29 +1,55 @@
...
-          resolveStateGo: function(stateName) {
-            _resolveStateGo(stateName);
-          }
-        };
-      }
-    ]);
+  angular
+  .module('source.router')
+  /**
+   * @namespace $routerProvider
```
Comment (Atap23) [inline source/router/source.router.factory.js]: Done

File: source/router/source.router.factory.js [hunk 10]
```diff
@@ -1,29 +1,55 @@
...
+  function $router($state, $timeout) {
+
+    return {
+      $state: $state,
+      resolveStateGo: resolveStateGo
+    };
+
+    /**
+     * @name _resolveStateGo
+     * @memberof source.router.$routerProvider
```
Comment (Atap23) [inline source/router/source.router.factory.js]: Done

File: source/router/source.router.factory.js (hunk 7 above)
Comment (Atap23) [inline source/router/source.router.factory.js]: Done

File: source/router/source.router.service.js (hunk 8 above)
Comment (Atap23) [inline source/router/source.router.service.js]: Done

**Pull request merged**